test/suite/test_io_capacity01.py

```diff
@@ -0,0 +1,128 @@
+
+
+    def generate_random_string(self, length):
+        characters = string.digits
+        random_string = ''.join(random.choice(characters) for _ in range(length))
+        return random_string
+
+    def get_stat(self, stat):
+        stat_cursor = self.session.open_cursor('statistics:')
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    # The number of written bytes exceeded the threshold.
+    def test_io_catacity_written_threshold(self):
+        # Close the initial connection. We will be opening new connections for this test.
+        self.close_conn()
+
+        #For a total of 1024 data writes of 1024 bytes each, the total number of bytes written is exactly 1M.
+        #The reconcile operation will add some additional data to ensure that the "total=1M" is triggered
+        insert_count = 1024
+        value_size = 1024
+        random_string = self.generate_random_string(value_size)
+
+        conn = self.wiredtiger_open(self.home, self.open_config)
+        self.session = conn.open_session()
+        self.session.create(self.uri, self.collection_cfg)
+        cursor = self.session.open_cursor(self.uri)
+        for i in range(insert_count):
+            cursor.set_key(i)
+            cursor.set_value(random_string)
+            cursor.insert()
+
+        #Take a checkpoint to ensure that the data is written to the disk
+        self.session.checkpoint('force=true')
+
+        self.assertGreater(self.get_stat(stat.conn.capacity_bytes_written), 0)
+        #background fsync statistics
+        self.assertGreater(self.get_stat(stat.conn.fsync_all_fh_total), 0)
+
+        self.session.create(self.uri, self.collection_cfg)
+        self.session.compact(self.uri)
+        self.session.close()
+
+    # The number of written bytes not exceeded the threshold, but the running period is exceeded
+    def test_io_catacity_fsync_background_period(self):
+        # Close the initial connection. We will be opening new connections for this test.
+        self.close_conn()
+
+        #only insert 1 data, the write bytes is well below 1M
+        #If the written conditions are not met fsync_maximum_wait_period, force background fsync
+        insert_count = 1
+        value_size = 1024
+        random_string = self.generate_random_string(value_size)
+        conn = self.wiredtiger_open(self.home, self.open_config)
+
+        self.session = conn.open_session()
+
+        self.session.create(self.uri, self.collection_cfg)
+        cursor = self.session.open_cursor(self.uri)
+        for i in range(insert_count):
+            cursor.set_key(i)
+            cursor.set_value(random_string)
+            cursor.insert()
+
+        #Take a checkpoint to ensure that the data is written to the disk
+        self.session.checkpoint('force=true')
+
+        time.sleep(self.fsync_time)
+        self.assertGreater(self.get_stat(stat.conn.capacity_bytes_written), 0)
```

Comment:
It would be more robust to spin on the statistic rather than sleeping.

Reply:
hi, @korteland 
how to spin, please guide me, I haven't used this before. To be honest, I'm not very familiar with Python.
thanks.

Reply:
Hi @y123456yz , no problem. So rather than sleeping, I would have some code like this:

```
while (self.get_stat(stat.conn.capacity_bytes_written) == 0):
    attempts += 1
    time.sleep(0.1)
    if attempts > 10000:
        throw Exception("Timed out waiting for statistic")
```

The numbers are all made up, and you should clean it up and experiment to see what works. But this is the overall idea.

Reply:
@korteland 
got it, thanks.
:)